crates/bevy_input/src/keyboard.rs

```diff
@@ -43,39 +106,61 @@ pub struct KeyboardInput {
 ///
 /// ## Differences
 ///
-/// The main difference between the [`KeyboardInput`] event and the [`Input<KeyCode>`] or [`Input<ScanCode>`] resources is that
+/// The main difference between the [`KeyboardInput`] event and the [`Input<KeyCode>`] resources is that
 /// the latter have convenient functions such as [`Input::pressed`], [`Input::just_pressed`] and [`Input::just_released`].
 pub fn keyboard_input_system(
-    mut scan_input: ResMut<Input<ScanCode>>,
     mut key_input: ResMut<Input<KeyCode>>,
     mut keyboard_input_events: EventReader<KeyboardInput>,
 ) {
     // Avoid clearing if it's not empty to ensure change detection is not triggered.
-    scan_input.bypass_change_detection().clear();
     key_input.bypass_change_detection().clear();
     for event in keyboard_input_events.read() {
         let KeyboardInput {
-            scan_code, state, ..
+            key_code, state, ..
         } = event;
-        if let Some(key_code) = event.key_code {
-            match state {
-                ButtonState::Pressed => key_input.press(key_code),
-                ButtonState::Released => key_input.release(key_code),
-            }
-        }
         match state {
-            ButtonState::Pressed => scan_input.press(ScanCode(*scan_code)),
-            ButtonState::Released => scan_input.release(ScanCode(*scan_code)),
+            ButtonState::Pressed => key_input.press(*key_code),
+            ButtonState::Released => key_input.release(*key_code),
         }
     }
 }
 
+/// Contains the platform-native physical key identifier
+///
+/// The exact values vary from platform to platform (which is part of why this is a per-platform
+/// enum), but the values are primarily tied to the key's physical location on the keyboard.
+///
+/// This enum is primarily used to store raw keycodes when Winit doesn't map a given native
+/// physical key identifier to a meaningful [`KeyCode`] variant. In the presence of identifiers we
+/// haven't mapped for you yet, this lets you use use [`KeyCode`] to:
+///
+/// - Correctly match key press and release events.
+/// - On non-web platforms, support assigning keybinds to virtually any key through a UI.
+#[derive(Debug, Clone, Ord, PartialOrd, Copy, PartialEq, Eq, Hash, Reflect)]
+#[cfg_attr(
+    feature = "serialize",
+    derive(serde::Serialize, serde::Deserialize),
+    reflect(Serialize, Deserialize)
+)]
+pub enum NativeKeyCode {
+    /// Unidentified
+    Unidentified,
+    /// An Android "scancode".
+    Android(u32),
```

Comment:
Why can I have a `NativeKeyCode::Android` on windows/linux/macos? shouldn't this be behind a `cfg` feature flag?

Reply:
I'm not sure ; maybe it would be helpful for some cross-platform communication ? It's a bit far fetched.

As that code is a copy/paste from winit, this conversation would be more fruitful on winit's repository.

Reply:
I opened an issue: https://github.com/rust-windowing/winit/issues/3238